src/main/webapp/script.js

```diff
@@ -86,12 +85,27 @@ function initMap() {
     curLocElement.innerText = place.geometry.location.toString();
 
     var url = new URL('/places-list', window.location.origin),
-        params = {lat: place.geometry.location.lat(), lng: place.geometry.location.lng()}
+        params = {
+          lat: place.geometry.location.lat(),
+          lng: place.geometry.location.lng(),
+          filter: getDieteryRestrictions()
+        }
     Object.keys(params).forEach(key => url.searchParams.append(key, params[key]))
     fetch(url).then((response => {
       response.json().then((data) => {
         getRandomRestaurant(data.results);
       })
     }))
   });
-}
\ No newline at end of file
+}
+
+function getDieteryRestrictions() {
+   const filterForm = document.getElementsByClassName('restaurant-filter');
```

Comment:
I found this interesting article on iterating DOM elements (in our case `HTMLCollection`), and introduced a couple of methods to use higher order methods (e.g. `filter()`) on `HTMLCollection`.

https://medium.com/@chuckdries/traversing-the-dom-with-filter-map-and-arrow-functions-1417d326d2bc

Reply:
Just sharing the article btw, current implementation is fine!

Reply:
Thanks for sharing! Currently I've made it so that it's only a single element, so this probably won't be applicable anymore unfortunately.